### ISSUE TITLE
Docker: --network-alias support

### ISSUE THREAD

**Issue by passcod** [hashicorp/nomad#1948]
In user-defined networks the `--network-alias` can be useful to manage service discovery within e.g. Docker overlay networks. Two containers having the same Network Alias results in multiple addresses listed under that alias.

E.g. a scheme we have currently under Nomad 0.4.0 (done by building our own `raw_exec`-based driver) is to give each container four aliases like this:

```
n${ALLOC_INDEX}.${TASK_NAME}.${GROUP_NAME}.${JOB_NAME}
${TASK_NAME}.${GROUP_NAME}.${JOB_NAME}
${GROUP_NAME}.${JOB_NAME}
${JOB_NAME}
```

such that we can easily route traffic to any single container, or to a random container in a task or group. (The last alias is not particularly useful for service discovery, but is required so that DNS queries resolve for its subdomains.)

Nomad 0.5.0-rc1 has all the features we need to drop our custom driver, except support for Docker network aliases. I acknowledge that it is probably too late for this to be included in 0.5.0, but we can workaround. Nevertheless, it would be a useful feature.

**Comment by dadgar** [comment]
Hey, if you are comfortable with Go, would love a PR

**Comment by passcod** [comment]
I'll have a go, but it's been years…

**Comment by dadgar** [comment]
Fixed by #1980
